tests/test_transforms/test_encoders/test_mean_encoder_transform.py

```diff
@@ -114,12 +114,112 @@ def expected_ts_begin_nan_smooth_1() -> TSDataset:
 def expected_ts_begin_nan_smooth_2() -> TSDataset:
     df = generate_ar_df(start_time="2001-01-01", periods=6, n_segments=1)
     df.rename(columns={"target": "mean_encoded_regressor"}, inplace=True)
-    df["mean_encoded_regressor"] = [np.NaN, np.NaN, 2 / 3, 5 / 4, 5 / 3, 2.5]
+    df["mean_encoded_regressor"] = [np.NaN, np.NaN, np.NaN, 5 / 3, 5 / 3, 2.5]
 
     ts = TSDataset(df, freq="D")
     return ts
 
 
+@pytest.fixture
+def multiple_nan_target_new_category_ts() -> TSDataset:
+    df = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=5)
+    df["target"] = [np.NaN, 1.5, np.NaN, 3.0, 4.0]
+
+    df_exog = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=6)
+    df_exog.rename(columns={"target": "regressor"}, inplace=True)
+    df_exog["regressor"] = ["A", "B", "A", "A", "B", "C"]
+
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future="all")
+
+    return ts
+
+
+@pytest.fixture
+def expected_multiple_nan_target_new_category_ts() -> TSDataset:
+    df = generate_ar_df(n_segments=1, start_time="2001-01-01", periods=5)
+    df.rename(columns={"target": "regressor_mean"}, inplace=True)
+    df["regressor_mean"] = [np.NaN, np.NaN, np.NaN, np.NaN, 1.5]
+
+    ts = TSDataset(df=df, freq="D")
+
+    return ts
+
+
+@pytest.fixture
+def multiple_nan_target_old_category_ts() -> TSDataset:
```

Comment:
What is the difference with `multiple_nan_target_new_category_ts`? How is it old vs. new?

Reply:
There are 2 main corner cases
1. If there is None in `target` for a new category where there were no notna `targets` yet, then the transform value should be NaN (in both cumcount and cumsum) (new)
2. If there is None in `target` for a category where there was already a notna `target`, then the transform value must be filled in with the previous one for this category (in both cumcount and cumsum) (old)

Reply:
It can't be understood from name of the tests. Can't we make them more obvious or at least write a description to them?

Reply:
For me now it seems like `expected_multiple_nan_target_new_category_ts` is just a subset of `expected_multiple_nan_target_old_category_ts`.